doc/tutorials/odbc-backend.md

```diff
@@ -0,0 +1,395 @@
+
+This tutorial describes how to set up unixODBC on Linux and use Elektra to retrieve configuration data from an _SQLite_ or _PostgreSQL_ database.
+
+> Currently, the _backend_odbc_ plugin is marked as EXPERIMENTAL and only supports reading data from data sources.
+> Writing data (e.g. with `kdb set`) should be supported soon.
+> Also, currently only data sources that define a table for _metadata_ are supported.
+> In the future, we plan to also support data sources without metadata tables.
+>
+> Please be aware that for using metadata, _outer joins_ have to be supported by the ODBC driver.
+
+> The ODBC backend plugin was tested with unixODBC, but should also work with iODBC and Microsoft ODBC (on Windows).
+> If you are using such an environment, feel free to share you experiences at https://issues.libelektra.org or extend this tutorial.
+
+> This tutorial uses SQLite- and PostgreSQL databases as data sources, but ODBC drivers for other data sources are also supported.
+
+## Overview
+
+The tutorial covers the following steps:
+
+1. Introduction
+2. Setting up the Databases for Configuration Data
+3. Installing unixODBC
+4. Installing the SQLite and PostgreSQL ODBC Drivers
+5. Creating the ODBC data sources for the Databases
+6. Compiling Elektra with support for the ODBC Backend
+7. Mounting the data sources into the global Key Database (KDB) of Elektra
+
+## 1. Introduction
+
+### Database scheme
+
+The basic database scheme is quite simple.
+You just need a table that contains two columns which can store a text of arbitrary length.
+Usually, the SQL data type `TEXT` is used for that purpose.
+One column is used for storing the **names** of the keys, the other one for storing their **values**.
+The column where the key-names are stored should be defined as the _primary key_ (PK) for that table.
+
+<!-- /* TODO: uncomment when support for data sources without a metadata table is implemented */
+It is _highly_ recommended, that you also define a table for storing _metadata_.
+Otherwise, lots of Elektra's features are not supported.-->
+
+Next, we have to create a table for storing metadata.
+The table for the metadata needs at least three columns:
+
+- A column with the _key-name_ which must be defined as a _foreign key_ (FK) to the PK of the table where the keys are stored.
+- A column for the _name_ of the metakeys.
+- A column for the _value_ of the metakeys.
+
+> The column for the **key-name** and the column for the **metakey-name** together form the PK of that table.
+
+Again, all these columns should be defined to store the data type `TEXT`.
+If you want to use `CHAR` or `VARCHAR` columns, it's your responsibility to make sure that the stored values don't exceed the maximum defined length.
+
+It is allowed that the tables contain more columns, these additional columns are not processed by Elektra,
+_must not_ be part of a PK and _must_ support NULL-values (if you want to add new keys in a future version of Elektra).
```

Comment:
> If multiple rows contain the same key-name column, they are treated as one key.
The value would be taken from the first key, the metadata of the two keys would be appended.

Please add this to the docs.

Reply:
Ok, I've added it, but also mentioned that this behavior is considered a part of the internal algorithm of the plugin and could change in the future.
So to be sure to get predictable and reasonable results, it's strongly advised to respect the PK and FK constraints.

Reply:
> that this behavior is considered a part of the internal algorithm of the plugin and could change in the future

That's fine, I'd also have accepted if the docs just explicitly state e.g. "duplicate key names are undefined behaviour". Just the implicitly undefined behaviour made it unclear whether there is or isn't consistent handling for this or if it could even lead to crashes/errors or corrupted data.